analyses/sex-prediction-from-RNASeq/03-evaluate_model.R

```diff
@@ -0,0 +1,163 @@
+    default = NULL,
+    help = "A character vector that will be used to name output files"
+  ),
+  optparse::make_option(
+    c("-s", "--seed"),
+    type = "integer",
+    default = 36354,
+    help = "seed integer",
+    metavar = "integer"
+  ),  
+  optparse::make_option(
+    c("-p", "--transcript_tail_percent"),
+    type = "double",
+    default = 0.25,
+    help = "Filter out the bottom (1 - transcript_tail_percent) of transcripts from the training set",
+    metavar = "double"
+  )
+  
+)
+
+
+# Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+#--------
+
+#--------test for the existence of the expected training and target value sets. 
+
+test_expression_file_name <- file.path(opt$test_set_input_directory, paste(opt$filename_lead, opt$seed, "test_expression.RDS", sep = "_"))
+test_targets_file_name <- file.path(opt$test_set_input_directory, paste(opt$filename_lead, opt$seed, "test_targets.tsv", sep = "_"))
+
+# print error and quit if not found.
+if (!file.exists(test_expression_file_name)) {
+  stop(paste("test set ", test_expression_file_name, "does not exist. Check all arguments."))
+}
+
+print("test set found!")
+
+if (!file.exists(test_targets_file_name)) {
+  stop(paste("targets set", test_targets_file_name, "does not exist. Check all arguments.", sep=" "))
+}
+
+print("targets set found!")
+
+# safe to read if we get here
+
+test_expression <- readRDS(test_expression_file_name)
+test_targets <- read.delim(test_targets_file_name, header=TRUE, sep="\t", stringsAsFactors = FALSE)
+
+#--------
+
+#--------test for the existence of the expected model and transcript index files
+
+model_file_name <- file.path(opt$model_input_directory, paste(opt$filename_lead, opt$seed, opt$transcript_tail_percent, "model_object.RDS", sep = "_"))
+transcript_index_file_name <- file.path(opt$model_input_directory, paste(opt$filename_lead, opt$seed, opt$transcript_tail_percent, "model_transcripts.RDS", sep = "_"))
+
+# print error and quit if not found.
+if (!file.exists(model_file_name)) {
+  stop(paste("model file ", model_file_name, "does not exist. Check all arguments."))
+}
+
+print("model file found!")
+
+if (!file.exists(transcript_index_file_name)) {
+  stop(paste("transcript index file", transcript_index_file_name, "does not exist. Check all arguments.", sep=" "))
+}
+
+print("transcript index set found!")
+
+# safe to read if we get here
+
+best_fit <- readRDS(model_file_name)
+model_transcripts <- readRDS(transcript_index_file_name)
+
+#--------
+
+#-------- predict from model and evaluate sex predictions with caret::confusionMatrix
+
+predict_class <- predict(best_fit, newx = test_expression[, model_transcripts], type = "class", s = best_fit$lambda.1se)
+predict_prob <- predict(best_fit, newx = test_expression[, model_transcripts], type = "response", s = best_fit$lambda.1se)
+
+cm_set <- data.frame(obs = test_targets[, "reported_gender"])
```

Comment:
We are going to want to evaluate performance in predicting the labels `germline_sex_estimate` at some point. There are a few ways we could approach this -- we could accept the target column name for what we want to use as the reference for the confusion matrix as a command line argument. So if you wanted to evaluate performance on both `reported_gender` and `germline_sex_estimate`, you would run this script twice. Alternatively you could hardcode the evaluation of both in this script. If we think about this as a general framework that will be adapted/built upon for future work (e.g., predicting disease labels), I think the first way lends itself to being a bit more flexible?

Reply:
I agree, and I am able to make this change today (Wednesday).

Reply:
Don't we need to specify the target column in the 02 script so that the model will be trained properly?

Reply:
My understanding of what is laid out on #84 is that we'll always train on `reported_gender`, but you could also change the 02 script to accept the correct column as a command line argument and set the default to `reported_gender`?